pkg/controller.v1alpha1/xgboost/controller.go

```diff
@@ -296,8 +310,19 @@ func (xc *XGBoostController) satisfiedExpectations(job *v1alpha1.XGBoostJob) boo
 	return false
 }
 
+// reconcilePyTorchJobs checks and updates replicas for each given PyTorchReplicaSpec.
+// It will requeue the job in case of an error while creating/deleting pods/services.
+func (xc *XGBoostController) reconcileXGBoostJobs(
```

Comment:
Can we reuse the reconciliation function from common controller?

Reply:
Lets try to reuse common controller code for new operators.